Dockerfile

```diff
@@ -1,29 +1,41 @@
-FROM ubuntu:16.04
+FROM ubuntu:18.04
 WORKDIR /ardupilot
 
-RUN useradd -U -d /ardupilot ardupilot && \
+RUN useradd -U -m ardupilot && \
     usermod -G users ardupilot
 
 RUN DEBIAN_FRONTEND=noninteractive apt-get update && apt-get install --no-install-recommends -y \
     lsb-release \
     sudo \
-    software-properties-common \
-    python-software-properties
+    software-properties-common
 
+COPY Tools/environment_install/install-prereqs-ubuntu.sh /ardupilot/Tools/environment_install/
+
+# Create non root user for pip
 ENV USER=ardupilot
-ADD . /ardupilot
 
 RUN echo "ardupilot ALL=(ALL) NOPASSWD:ALL" > /etc/sudoers.d/ardupilot
 RUN chmod 0440 /etc/sudoers.d/ardupilot
 
 RUN chown -R ardupilot:ardupilot /ardupilot
 
 USER ardupilot
-RUN bash -c "Tools/environment_install/install-prereqs-ubuntu.sh -y"
-RUN sudo apt-get clean \
+
+ENV SKIP_AP_EXT_ENV=1 SKIP_AP_GRAPHIC_ENV=1 SKIP_AP_COV_ENV=1 SKIP_AP_GIT_CHECK=1
+RUN Tools/environment_install/install-prereqs-ubuntu.sh -y
+
+# add Tools to bashrc manually as docker don't load the .profile
```

Comment:
Can we pass in a file to tack all this crap onto and use it in the env script?

Reply:
yep using the .bashrc is working 
Otherwise we could do as I have done for completion a just have to source an env file on ardupilot directory and manage the environment addition in this file